providers/blacklist_provider.rego

```diff
@@ -0,0 +1,26 @@
+package terraform
+
+import input.tfplan as tfplan
+
+# Blacklisted Terraform providers
+not_allowed_provider = [
+  "azurerm"
```

Comment:
@rtfee jfyi, interesting fact: starting from Terraform v0.13 provider_name in plan json will be stored as fully qualified source name (e.g. `registry.terraform.io/hashicorp/azurerm`). We'll handle this in SCALRCORE-16889.